File: app/src/main/java/com/damhoe/scoresheetskat/score/domain/SkatResult.java
package com.damhoe.scoresheetskat.score.domain;

public enum SkatResult {
 WON,
 LOST,
 OVERBID,
 PASSE;

 public int asInteger() {
  return ordinal();
 }

 public static SkatResult fromInteger(int value) {
  for (SkatResult result: SkatResult.values()) {
   if (result.ordinal() == value) {
    return result;
   }
  }
  return SkatResult.PASSE; // default
 }
}

File: app/src/main/java/com/damhoe/skatscores/score/domain/SkatResult.kt
package com.damhoe.skatscores.score.domain

enum class SkatResult {
    WON, LOST, OVERBID, PASSE;

    fun asInteger() = ordinal

    companion object {
        fun fromInteger(value: Int): SkatResult {
            for (result in values()) {
                if (result.ordinal == value) {
                    return result
                }
            }
            return PASSE // default
        }
    }
}